packages/react/src/select/item/SelectItem.tsx

```diff
@@ -125,7 +128,13 @@ export const SelectItem = React.forwardRef(function SelectItem(
 ) {
   const { value: valueProp = null, label, ...otherProps } = props;
 
-  const listItem = useCompositeListItem({ label });
+  const itemTextRef = React.useRef<HTMLElement | null>(null);
+
+  const getItemText = React.useCallback(() => {
+    return itemTextRef.current?.textContent ?? null;
+  }, []);
+
+  const listItem = useCompositeListItem({ label, getItemText });
```

Comment:
Isn't this logic duplicated with the `.textContent` read in `useCompositeListItem()`?